### PR TITLE
feat(forms): add novalidate by default

### LINKED ISSUE #12201: Add novalidate by default to forms with FormsModule or ReactiveFormsModule
Should be able to turn off default behavior with `ngNoForm`.

cc @IgorMinar

### PR DESCRIPTION
**Pull request by DzmitryShylovich**
Fixes https://github.com/angular/angular/issues/12201

cc @kara

### CODE REVIEW

**Review by kara — APPROVED** [review]
LGTM

**Comment by gkalpak** [comment]
Isn't this:

1. a breaking change?
2. making it impossible to use overwrite "the default" (i.e. enable native validate behavior, while still having Angular forms functionality)?

cc @kara

**Comment by vicb** [comment]
> a breaking change?

As a feat it will only be merged in the 4.x branch

File: modules/@angular/forms/src/directives/ng_novalidate_directive.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Directive} from '@angular/core';
+
+@Directive({
+  selector: 'form:not([ngNoForm])',
+  host: {'novalidate': ''},
+})
+export class NgNovalidate {
```
Comment (vicb) [inline modules/@angular/forms/src/directives/ng_novalidate_directive.ts]: @DzmitryShylovich Could you please add docs here (as part of the other PR).

**Comment by gkalpak** [comment]
> As a feat it will only be merged in the 4.x branch

Still, shouldn't breaking changes be mentioned in the Changelog? Or maybe a migration guide (if there is going to be one)?

Also, the fact that this is _"making it impossible to use overwrite "the default" (i.e. enable native validate behavior, while still having Angular forms functionality)"_ seems quite bold. Was it intentional?

**Comment by vicb** [comment]
@gkalpak I have asked @DzmitryShylovich to open an issue on angular/angular.io for the docs.

A migration guide would actually be a good idea /cc @wardbell @juleskremer 

> Also, the fact that this is "making it impossible to use overwrite "the default" (i.e. enable native validate behavior, while still having Angular forms functionality)" seems quite bold. Was it intentional?

@DzmitryShylovich has submitted a PR to fix this, see #13573 linked above

**Comment by gkalpak** [comment]
Aha, I missed #13573. Thx @vicb, @DzmitryShylovich 👍

**Pull request merged**